src/texture.rs

```diff
@@ -42,24 +42,38 @@ pub fn load_texture(bytes: &[u8], display: &Display<WindowSurface>) -> Texture2d
 /// returns a Texture2dArray with these images.
 /// ImageCut is in the format \[x, y, height, width\] (u,v) coord,
 /// (0, 0) is top left and x, y, height and width are in fraction of the image.
-pub fn load_texture_cut(bytes: &[u8], display: &Display<WindowSurface>, cut: &[ImageCut]) -> Texture2dArray {
-    let image = image::load(std::io::Cursor::new(bytes),
-                            image::ImageFormat::Png).unwrap().to_rgba8();
-    let mut source = Vec::new();
-    // Set a scaling factor which is a common multiplier for every texture
+pub fn load_texture_cut(root : &str, all_textures_name: Vec<&str>, display: &Display<WindowSurface>, cut: &[ImageCut]) -> Texture2dArray {
```

Comment:
Now that I am looking into this code, I realize that this is actually quite a big problem... It's so easy to not use correctly the instantiation of entities. It's also hard to understand that is happening. 

I guess that during the weekend I did not see this so well and left it go... The huge problem here is that there are hard-coded `u8` a bit everywhere, with no compile-time safety.

So here's what I suggest. It's a refactoring of the entities' textures managment.

- First we will focus on `PLAYER_CUT_TEMPLATES` of type `&[ImageCut]`. This is the root of our problems. It is the position inside this array that encodes the body position. This is why the code seems so messy. So we will completly refactor its type. Let's create a new type: 

```rust
enum BodyPart {Leg, Head, <ETC> }

struct HumanoidTextureCut {
    parts: HashMap<BodyPart, [ImageCut; 6]>
}

impl HumanoidTextureCut {
    fn new(head: [ImageCut; 6], leg: [ImageCut; 6], <ETC...>) -> Self {
        TODO
     }
}
```

- The function `load_texture_cut` now receives `&HumanoidTextureCut` instead of `&[ImageCut]` and we can rewrite `PLAYER_CUT_TEMPLATES` using this struct.

==> benefits = the 'sementic' of knowing where each 'body part' is in the array is provided by the type system, and not by hard-coding `u8` in the code.

- The problem remains that the function that creates the `EntityCube::new()` (`get_opengl_entities`) needs to know properly which cube has which type. But the good news is that now we have a struct  `BodyPart`, so we can add a method `id(&self) -> u8` to `EntityCube`